src/cpp/tests/automation/testthat/test-automation-data-viewer.R

```diff
@@ -0,0 +1,14 @@
+
+library(testthat)
+
+self <- remote <- .rs.automation.createRemote()
+client <- remote$client
+
+# https://github.com/rstudio/rstudio/pull/14657
+test_that("we can use the data viewer with temporary R expressions", {
```

Comment:
Not sure if there's a more appropriate test, but at least this exercises the underlying code.